src/jaxsim/integrators/common.py

```diff
@@ -0,0 +1,503 @@
+import abc
+import dataclasses
+from typing import Any, ClassVar, Generic, Protocol, Self, Type, TypeVar
```

Comment:
This ensures partial compatibility with Python 3.10, which is nice to have at this stage

```suggestion
from typing import Any, ClassVar, Generic, Protocol, Type, TypeVar

try:
    from typing import Self
except ImportError:
    from typing_extensions import Self
```

Reply:
We ditched on purpose Python 3.10, in fact you can see:

https://github.com/ami-iit/jaxsim/blob/389c556de16308505d65ae9f38e7dd74108fdb4d/setup.cfg#L53

Reply:
Yes, but for some internal lab activities, I'm still using it with python 3.10 as some libraries are still not compatible with python 3.11. In addition, I've seen the same logic in other parts of the code: https://github.com/ami-iit/jaxsim/blob/389c556de16308505d65ae9f38e7dd74108fdb4d/tests/test_jax_oop.py#L14-L17, https://github.com/ami-iit/jaxsim/blob/389c556de16308505d65ae9f38e7dd74108fdb4d/src/jaxsim/utils/vmappable.py#L10-L13, https://github.com/ami-iit/jaxsim/blob/389c556de16308505d65ae9f38e7dd74108fdb4d/src/jaxsim/utils/jaxsim_dataclass.py#L14-L17, https://github.com/ami-iit/jaxsim/blob/389c556de16308505d65ae9f38e7dd74108fdb4d/src/jaxsim/simulation/simulator.py#L6-L9
wouldn't it be better to unify it everywhere or delete it everywhere?

Reply:
In this case, it makes sense to add -unofficial- Python 3.10 support. It's minor changes, for now. I expect that we migrate to Ubuntu 24.04 sooner or later, with Python 3.11 as default.

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/88/commits/4e49b171157c270b5c4d17bb04de5ba589b51232